src/lib.rs

```diff
@@ -202,21 +245,45 @@ impl PerformanceData {
             SetTimeFlags::Default,
         );
         while current <= end {
+            aperf_collect_data.time = TimeEnum::DateTime(Utc::now());
+            aperf_collect_data.data = HashMap::new();
             let ret = tfd.read();
             if ret > 1 {
                 error!("Missed {} interval(s)", ret - 1);
             }
             debug!("Time elapsed: {:?}", start.elapsed());
             current += time::Duration::from_secs(ret * self.init_params.interval);
-            for (_name, datatype) in self.collectors.iter_mut() {
+            for (name, datatype) in self.collectors.iter_mut() {
                 if datatype.is_static {
                     continue;
                 }
-                datatype.collect_data()?;
-                datatype.write_to_file()?;
+
+                aperf_collect_data.measure(
+                    name.clone() + "-collect",
+                    |d: &mut data::DataType| -> Result<()> {
+                        d.collect_data()?;
+                        Ok(())
+                    },
+                    datatype,
+                )?;
+                aperf_collect_data.measure(
+                    name.clone() + "-print",
+                    |d: &mut data::DataType| -> Result<()> {
+                        d.write_to_file()?;
+                        Ok(())
+                    },
+                    datatype,
+                )?;
```

Comment:
I think this should be able to be
```
aperf_collect_data.measure(
    name.clone() + "-print",
    || -> Result<()> {
        d.write_to_file()
    }
);
```
if measure() is made to take a no-args function. I guess it would still have to be typed to return Result<()> since we do have to keep failure handling working